pkg/backup/backup.go

```diff
@@ -730,3 +764,32 @@ type tarWriter interface {
 	Write([]byte) (int, error)
 	WriteHeader(*tar.Header) error
 }
+
+func updateVolumeInfos(
+	volumeInfos []*volume.VolumeInfo,
+	unstructured unstructured.Unstructured,
+	groupResource schema.GroupResource,
+	log logrus.FieldLogger,
+) {
+	switch groupResource.String() {
+	case kuberesource.DataUploads.String():
+		var dataUpload velerov2alpha1.DataUpload
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructured.UnstructuredContent(), &dataUpload)
+		if err != nil {
+			log.WithError(err).Errorf("fail to convert DataUpload: %s/%s",
+				unstructured.GetNamespace(), unstructured.GetName())
+		}
+
+		for index := range volumeInfos {
+			if volumeInfos[index].PVCName == dataUpload.Spec.SourcePVC &&
+				volumeInfos[index].PVCNamespace == dataUpload.Spec.SourceNamespace {
+				if dataUpload.Status.CompletionTimestamp != nil {
+					volumeInfos[index].CompletionTimestamp = dataUpload.Status.CompletionTimestamp
+				}
+				volumeInfos[index].SnapshotDataMovementInfo.SnapshotHandle = dataUpload.Status.SnapshotID
+				volumeInfos[index].SnapshotDataMovementInfo.RetainedSnapshot = dataUpload.Spec.CSISnapshot.VolumeSnapshot
+				volumeInfos[index].SnapshotDataMovementInfo.Size = dataUpload.Status.Progress.TotalBytes
+			}
```

Comment:
Missing `StartTimeStamp`

Reply:
VolumeInfos generation already fills StartTimestamp, so there is no need to overwrite it here.